numpy/lib/format.py

```diff
@@ -317,12 +317,14 @@ def _write_array_header(fp, d, version=None):
     if len(header) >= (256*256) and version == (1, 0):
         raise ValueError("header does not fit inside %s bytes required by the"
                          " 1.0 format" % (256*256))
-    if len(header) < (256*256):
+    elif len(header) < (256*256):
```

Comment:
How is this? With `(1, 0)` as the default `version` instead of `None` things are simpler.